arPullGlob.py

```diff
@@ -22,108 +24,71 @@
 
 def makList():
     #################
-    # this function generates a PV list for cavities
-    # in SC linac
+    # this function generates a PV prefix list for cavities
+    # in SC linac like 'ACCL:L0B:0110:'
     #################
 
     pvList = []
-    #cavPv = []
-    for lin in range(4):     #  for the 4 linac sections
-        if lin == 0:         #  start with L0B
-            for j in range(1, 9):
-                pvList.append("ACCL:L" + str(lin) + "B:01" + str(j))
-        if lin == 1:         # move to L1B and do two 1.3GHz CM
-            for n in range(2, 4):
-                for j in range(1, 9):
-                    pvList.append("ACCL:L" + str(lin) + "B:" + "0" + str(n) + str(j))
-                             # then do the H1 and H2 3.9 GHz CM
-            for n in range(1, 3):
-                for j in range(1, 9):
-                    pvList.append("ACCL:L" + str(lin) + "B:" + "H" + str(n) + str(j))
-                             # move on to the 12 CMs in L2B
-        if lin == 2:
-            for n in range(4, 16):
-                if n < 10:
-                    for j in range(1, 9):
-                        pvList.append("ACCL:L" + str(lin) + "B:" + "0" + str(n) + str(j))
-                             # remember to adjust CM number in PV if greater than '10'
-                else:
-                    for j in range(1, 9):
-                        pvList.append("ACCL:L" + str(lin) + "B:" + str(n) + str(j))
-                             # Finally, do the 20 CMs in L3B
-        if lin == 3:
-            for n in range(16, 36):
-                for j in range(1, 9):
-                    pvList.append("ACCL:L" + str(lin) + "B:" + str(n) + str(j))
-
+    for lo in LINAC_OBJECTS:
+        for cm in lo.cryomodules:
+            for cav in lo.cryomodules[cm].cavities:
+                pvList.append(lo.cryomodules[cm].cavities[cav].pvPrefix)
     return pvList;
 
 
-
-class Archiver(object):
-
-
-    def __init__(self, machine):
-        # type: (str) -> None
-        self.url_formatter = ARCHIVER_URL_FORMATTER.format(MACHINE=machine)
-        
-        #******************************************************
-
 #******************************************************
 
-    def getValuesOverTimeRange(self, strtTim, endTim, timeInterval=None):
+def getValuesOverTimeRange(strtTim, endTim, timeInterval=None):
  #       # type: (List[str], datetime, datetime, int) -> Dict[str, Dict[str, List[Union[datetime, str]]]]
-        global resultsFB, resultsIntlk, resultsRfRdy, startTime, endTime
-        pvList=[]
-        cavPvRfRdy={}
-        cavPvFB={}
-        cavPvIntlk={}
+    global resultsFB, resultsIntlk, resultsRfRdy, startTime, endTime
```

Comment:
Just curious about the global tag here. I think I would personally prefer things like `RESULTS_FB = None` at the top of the file if they're being treated as globals (vs being defined here)

Reply:
Agreed. Not sure why they're globals, trying to suss that out now.